src/module/migration.js

```diff
@@ -0,0 +1,112 @@
+                console.log(`Migrating Actor document ${actor.name}`);
+                await actor.update(updateData);
+            }
+        } catch(err) {
+            err.message = `Failed OSE System Migration for Actor ${actor.name}: ${err.message}`;
+            console.log(err);
+        }
+    }
+
+    // Set the migration as complete
+    game.settings.set(game.system.id, "systemMigrationVersion", game.system.version)
+    ui.notifications.info(game.i18n.format("OSE.migration.completeMigration", {version}), {permanent: true});
+};
+
+/* ------------------------------- */
+/* Document Type Migration Helpers */
+/* ------------------------------- */
+
+/**
+ * Migrate a single Actor document to implement latest data model changes.
+ * 
+ * @param {object} actor                The actor data object to update
+ * @param {object} [migrationData ]     Additional data to perform the mitigation
+ * @returns {object}                    The updateData to apply
+ */
+export const migrateActorData = function(actor, migrationData) {
+    const updateData = {};
+
+    // Migrate owned items
+    if ( !actor.items ) return updateData;
+    const items = actor.items.reduce((arr, i) => {
+        // Migrate owned items
+        const itemData = i instanceof CONFIG.Item.documentClass ? i.toObject() : i;
+        let itemUpdate = migrateItemData(itemData, migrationData);
+
+        // Update the Owned Item
+        if ( !foundry.utils.isEmpty(itemUpdate) ) {
+            itemUpdate._id = itemData._id;
+            arr.push(foundry.utils.expandObject(itemUpdate));
+        }
+
+        return arr;
+    }, []);
+    if ( items.length > 0 ) updateData.items = items;
+
+    return updateData;
+}
+
+
+/**
+ * Migrate a single item to implement latest data model changes
+ * 
+ * @param {object} item             Item data to migrate
+ * @param {object} [migrationData]  Additional data to perform the migration
+ * @returns {object}                The updateData to apply
+ */
+
+export function migrateItemData(item, migrationData) {
+    const updateData = {};
+    _migrateContainerItemIds(item, updateData, migrationData );
+
+    return updateData;
+}
+
+
+/* ----------------------------- */
+/* Low level migration functions */
+/* ----------------------------- */
+
+/**
+ * Converts faulty itemIds, where itemIds stored objects instead of ids.
+ * @param {object} document 
+ * @param {object} updateData
+ * @returns {object}
+ * @private
+ */
+function _migrateContainerItemIds(document, updateData, {actorId}={}) {
+    let itemIds = document?.system?.itemIds;
+    let containerId = document?._id;
+    
+    if (actorId && itemIds) {
+        const actor = game.actors.get(actorId);
+        const containerItems = actor.items.reduce((arr, i) => {
+            if ( i.system.containerId === containerId ) 
+                arr.push(i._id);
+
+            return arr;
+        }, []);
+
+        updateData.system = { itemIds: containerItems };
```

Comment:
Is there a way to do this update from the data model?

Reply:
I will investigate!

Reply:
I've looked into the data models but I am not sure how I would approach doing this update from there. Can you be a bit more specific with what you're thinking here?

Reply:
Since I need to access the actors item, I can't use `static migrateData` to perform this update, I did check that at least for the container data model.

Reply:
correct me if I wrong, but I think I saw yesterday there was a path @wyrmisis suggested that @bakbakbakbakbak is still investigating?

Reply:
I looked into it, but was unable to get access to `this.actor` at the `migrateData` step, and to find the `itemIds`, that is unfortunately necessary (as far as I can tell). I checked in the #system-development in the foundry-discord but didn't get any further insight.

Perhaps @wyrmisis has some further insight here?

Reply:
Got it working! The latest changes is migrating containers (checks if the same items that have said containers' `_id` are in the containers `system.itemIds` field, if not it modifies it).